libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt

```diff
@@ -15,31 +17,42 @@ import java.util.concurrent.ConcurrentLinkedQueue
  * @property context Context
  * @property accessToken String
  * @property language [Locale] language (ISO 639)
+ * @property options [VoiceInstructionsPlayerOptions] (optional)
  */
 @UiThread
-class MapboxVoiceInstructionsPlayer(
+class MapboxVoiceInstructionsPlayer @JvmOverloads constructor(
     private val context: Context,
     private val accessToken: String,
-    private val language: String
+    private val language: String,
+    private val options: VoiceInstructionsPlayerOptions = VoiceInstructionsPlayerOptions.Builder()
+        .build()
 ) : VoiceInstructionsPlayer {
 
     private val playCallbackQueue: Queue<PlayCallback> = ConcurrentLinkedQueue()
-    // for testing
-    internal var filePlayer: VoiceInstructionsFilePlayer =
-        VoiceInstructionsFilePlayer(context, accessToken, language)
-    // for testing
-    internal var textPlayer: VoiceInstructionsTextPlayer =
-        VoiceInstructionsTextPlayer(context, language)
+    private val filePlayer: VoiceInstructionsFilePlayer =
+        VoiceInstructionsFilePlayerProvider.retrieveVoiceInstructionsFilePlayer(
+            context,
+            accessToken,
+            language
+        )
+    private val textPlayer: VoiceInstructionsTextPlayer =
+        VoiceInstructionsTextPlayerProvider.retrieveVoiceInstructionsTextPlayer(context, language)
     private var localCallback: VoiceInstructionsPlayerCallback =
         object : VoiceInstructionsPlayerCallback {
             override fun onDone(state: SpeechState.DonePlaying) {
+                audioFocusDelegate.abandonFocus()
                 val currentPlayCallback = playCallbackQueue.poll()
                 val announcement = currentPlayCallback.announcement
```

Comment:
`currentPlayCallback` can be `null` here because `playCallbackQueue.poll()` returns null when the queue is empty.

Reply:
Although, the IDE warns you, that's not an option for us because only one announcement can be played at a time.

Reply:
I am just thinking if it is possible that we are trying to poll even though the queue is empty. Should we wrap all of this inside `if(playCallbackQueue.isNotEmpty())` 🤔

Reply:
It shouldn't be the case. Also, I'd say this way we'll find bugs much quicker as will showcase issues in the design.